src/overlays/actors/ovl_En_Baguo/z_en_baguo.c

```diff
@@ -449,9 +449,9 @@ void EnBaguo_InitializeEffect(EnBaguo* this, Vec3f* pos, Vec3f* velocity, Vec3f*
             effect->accel = *accel;
             effect->scale = scale;
             effect->timer = timer;
-            effect->rotation.x = (s16)randPlusMinusPoint5Scaled(30000.0f);
-            effect->rotation.y = (s16)randPlusMinusPoint5Scaled(30000.0f);
-            effect->rotation.z = (s16)randPlusMinusPoint5Scaled(30000.0f);
+            effect->rotation.x = (s16)Rand_CenteredFloat(30000.0f);
+            effect->rotation.y = (s16)Rand_CenteredFloat(30000.0f);
+            effect->rotation.z = (s16)Rand_CenteredFloat(30000.0f);
```

Comment:
I remember there was an issue casting f32 directly to s16. I've gone and added in `(s16)(s32)` casts to all these instead of a direct s16 cast

Reply:
That's the thing about UB from casting into a type that's too small, here it's fine since the number's within ±15k, which is well within an s16. I'm more surprised that casts are needed at all if it's going into an s16 anyway.